core-services/src/main/java/org/zalando/nakadi/service/publishing/EventPublisher.java

```diff
@@ -131,8 +126,11 @@ EventPublishResult processInternal(final String events,
             }
             validateEventOwnership(eventType, batch);
             validate(batch, eventType, delete);
+            if (PartitionStrategy.HASH_STRATEGY.equals(eventType.getPartitionStrategy())) {
+                setPartitionKeys(batch, eventType);
+            }
```

Comment:
call with `if PartitionStrategy == something: setPartitionKeys` right before calling `partition` method looks very strange. Does it make sense to put it to `paritition()` method?